tests/test_taskprocessor.c

```diff
@@ -963,6 +966,174 @@ AST_TEST_DEFINE(serializer_pool)
 	return AST_TEST_PASS;
 }
 
+/*!
+ * \brief Test for CLI command "core show taskprocessor <name>"
+ *
+ * This test creates a taskprocessor, queues tasks with controlled execution,
+ * and verifies that the CLI command displays the queued tasks correctly.
+ */
+AST_TEST_DEFINE(taskprocessor_cli_show)
+{
+	RAII_VAR(struct ast_taskprocessor *, tps, NULL, ast_taskprocessor_unreference);
+	struct task_data *task_data1 = NULL;
+	struct task_data *task_data2 = NULL;
+	struct task_data *task_data3 = NULL;
+	char cli_command[128];
+	int cli_output_fd[2];
+	char output_buffer[4096] = {0};
+	ssize_t bytes_read;
+	int res = AST_TEST_FAIL;
+
+	switch (cmd) {
+	case TEST_INIT:
+		info->name = "taskprocessor_cli_show";
+		info->category = "/main/taskprocessor/";
+		info->summary = "Test CLI command 'core show taskprocessor'";
+		info->description =
+			"Verifies that the 'core show taskprocessor <name>' CLI command\n"
+			"displays taskprocessor information and queued tasks correctly.";
+		return AST_TEST_NOT_RUN;
+	case TEST_EXECUTE:
+		break;
+	}
+
+	/* Create a pipe to capture CLI output */
+	if (pipe(cli_output_fd) != 0) {
+		ast_test_status_update(test, "Failed to create pipe for CLI output\n");
+		return AST_TEST_FAIL;
+	}
+
+	/* Create taskprocessor */
+	tps = ast_taskprocessor_get("test_cli_taskprocessor", TPS_REF_DEFAULT);
+	if (!tps) {
+		ast_test_status_update(test, "Unable to create test taskprocessor\n");
+		close(cli_output_fd[0]);
+		close(cli_output_fd[1]);
+		return AST_TEST_FAIL;
+	}
+
+	/* Create tasks that will wait so they stay in the queue */
+	task_data1 = task_data_create();
+	task_data2 = task_data_create();
+	task_data3 = task_data_create();
+
+	if (!task_data1 || !task_data2 || !task_data3) {
+		ast_test_status_update(test, "Unable to create task_data\n");
+		goto cleanup;
+	}
+
+	/* Set a long wait time so tasks stay queued */
+	task_data1->wait_time = 5000; /* 5 seconds */
+	task_data2->wait_time = 5000;
+	task_data3->wait_time = 5000;
+
+	/* Queue the tasks */
+	if (ast_taskprocessor_push(tps, task, task_data1)) {
+		ast_test_status_update(test, "Failed to queue task 1\n");
+		goto cleanup;
+	}
+	if (ast_taskprocessor_push(tps, task, task_data2)) {
+		ast_test_status_update(test, "Failed to queue task 2\n");
+		goto cleanup;
+	}
+	if (ast_taskprocessor_push(tps, task, task_data3)) {
+		ast_test_status_update(test, "Failed to queue task 3\n");
+		goto cleanup;
+	}
+
+	/* Give the taskprocessor a moment to start processing */
+	usleep(100000); /* 100ms */
```

Comment:
This feels like a race condition waiting to happen... does it actually need to wait?